grade-scripts/trackers-from-requests.js

```diff
@@ -93,53 +96,57 @@ const calculateTrackerPrevalence = () => {
-                let parent = entityList[trackerUrl] || network
-                networksOnThisSite[parent] = true
+            // get all the tracker networks seen on a specific site
+            Object.keys(siteData.trackersBlocked).forEach((network) => {
+                Object.keys(siteData.trackersBlocked[network]).forEach((trackerUrl) => {
+                    let parent = entityList[trackerUrl] || network
+                    networksOnThisSite[parent] = true
+                })
             })
-        })
-        Object.keys(siteData.trackersNotBlocked).forEach((network) => {
-            Object.keys(siteData.trackersNotBlocked[network]).forEach((trackerUrl) => {
-                let parent = entityList[trackerUrl] || network
-                networksOnThisSite[parent] = true
+            Object.keys(siteData.trackersNotBlocked).forEach((network) => {
+                Object.keys(siteData.trackersNotBlocked[network]).forEach((trackerUrl) => {
+                    let parent = entityList[trackerUrl] || network
+                    networksOnThisSite[parent] = true
+                })
             })
-        })
 
-        // add them to the global counts
-        Object.keys(networksOnThisSite).forEach((network) => {
-            if (network === 'unknown') return
+            // add them to the global counts
+            Object.keys(networksOnThisSite).forEach((network) => {
+                if (network === 'unknown') return
 
-            if (!networksSeen[network]) {
-                networksSeen[network] = 0
-            }
+                if (!networksSeen[network]) {
+                    networksSeen[network] = 0
+                }
 
-            networksSeen[network] += 1
+                networksSeen[network] += 1
+            })
         })
-    })
 
-    // calculate overall prevalence
-    let networkPrevalence = {}
-    let totalSites = siteDataArray.length
+        // calculate overall prevalence
+        let totalSites = siteDataArray.length
 
-    Object.keys(networksSeen).forEach((network) => {
-        let percent = networksSeen[network] / totalSites * 100
-        // round to 2 significant digits
-        percent = Math.round(percent * 100) / 100
+        Object.keys(networksSeen).forEach((network) => {
+            let percent = networksSeen[network] / totalSites * 100
+            // round to 2 significant digits
+            percent = Math.round(percent * 100) / 100
 
-        networkPrevalence[network] = percent
-    })
+            networkPrevalence[network] = percent
+        })
 
-    // write the prevalence to file, then add it to all the file data
-    fs.writeFileSync(`data/generated/prevalence.json`, JSON.stringify(networkPrevalence))
+        // write the prevalence to file, then add it to all the file data
+        fs.writeFileSync(`data/generated/prevalence.json`, JSON.stringify(networkPrevalence))
+    } else {
+        networkPrevalence = require(`../data/generated/prevalence.json`)
+    }
```

Comment:
The changes here are better viewed with whitespace disabled: https://github.com/duckduckgo/privacy-grade/pull/17/files?w=1

Reply:
This basically makes sure that, if we're not regenerating prevalence this time the script is run, we grab the file that already exists instead.

That way all the prevalence data is present in the tracker files.